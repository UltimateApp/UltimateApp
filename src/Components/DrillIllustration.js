import React, { useState, useEffect, useRef } from 'react';
import { Animated, View, StyleSheet, Text, Dimensions, TouchableOpacity, Image } from 'react-native';
import GestureRecognizer from 'react-native-swipe-gestures';
import { WebView } from 'react-native-webview';

import Animation from './animation/Animation';
import VimeoVideo from './VimeoVideo';
import { IllustrationType } from '../Fixtures';
import theme from '../styles/theme.style';
import { swipeConfig } from '../styles/config';
import iconRedo from '../../assets/redo_arrow.png';
import buttonValidation from '../../assets/button_validation_ultra_light.png';
import buttonValidationGradient from '../../assets/button_validation_gradient.png';
import { Easing } from 'react-native-reanimated';

const DrillIllustration = props => {
  const [currentStepIndex, setStepIndex] = useState(0);

  const opacityUnchecked = useRef(new Animated.Value(1)).current;
  const opacityChecked = opacityUnchecked.interpolate({ inputRange: [0, 1], outputRange: [1, 0] });

  const checkAnimation = () => {
    Animated.sequence([
      Animated.timing(opacityUnchecked, {
        toValue: 0,
        duration: 800,
        easing: Easing.easeOutQuint,
      }),
      Animated.timing(opacityUnchecked, {
        toValue: 1,
        duration: 10,
      }),
    ]).start(() => incrementStepIndex());
  };

  const checkAnimationFast = () => {
    Animated.sequence([
      Animated.timing(opacityUnchecked, {
        toValue: 0,
        duration: 200,
        easing: Easing.easeOutQuint,
      }),
      Animated.timing(opacityUnchecked, {
        toValue: 1,
        duration: 10,
      }),
    ]).start(() => incrementStepIndex());
  };

  const currentStep = props.drill.steps[currentStepIndex];

  // back to 0 when drill changes
  useEffect(() => {
    setStepIndex(0);
  }, [props.drill]);

  const incrementStepIndex = () => {
    setStepIndex((currentStepIndex + 1) % (props.drill.steps.length + 1));
  };

  const decrementStepIndex = () => {
    if (currentStepIndex === 0) {
      setStepIndex(0);
    } else {
      setStepIndex((currentStepIndex - 1) % (props.drill.steps.length + 1));
    }
  };

  const displayNextStep = () => {
    if (currentStepIndex + 1 === props.drill.steps.length) {
      return (
        <>
          <View>
            <TouchableOpacity onPress={() => checkAnimationFast()}>
              <View style={styles.description}>
                <View style={styles.wrapperFinish}>
                  <Text style={styles.fitnessNext}>Finish</Text>
                </View>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.lines} />
        </>
      );
    } else {
      return (
        <>
          <View>
            <TouchableOpacity onPress={() => checkAnimationFast()}>
              <View style={styles.description}>
                <View style={styles.subWrapper}>
                  <Text style={styles.fitnessNext}>{props.drill.steps[currentStepIndex + 1].repetition}</Text>
                </View>
                <View style={styles.subSubWrapper}>
                  <Text style={styles.fitnessNext}>{props.drill.steps[currentStepIndex + 1].title}</Text>
                </View>
                <View style={styles.fakeWrapper} />
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.lines} />
        </>
      );
    }
  };

  const checkSwitch = () => {
    if (currentStepIndex === props.drill.steps.length) {
      return displayFinish();
    } else if (!currentStep) {
      return <View />; // bad state, but let's not crash
    } else {
      switch (props.drill.steps[currentStepIndex].illustrationType) {
        case IllustrationType.ANIMATION:
          return displayAnimation(props.drill.steps[currentStepIndex]);
        case IllustrationType.YOUTUBE:
          return displayYoutube(props.drill.steps[currentStepIndex]);
        case IllustrationType.VIMEO:
          return displayVimeo(props.drill.steps[currentStepIndex]);
        default:
          return <Text>No visual content for this drill</Text>;
      }
    }
  };

  const displayFinish = () => {
    return (
      <>
        <View style={styles.containerFinish}>
          <Text style={styles.redoMessage}>You have completed the drill!</Text>
          <TouchableOpacity style={styles.redoButton} onPress={() => incrementStepIndex()}>
            <Image style={styles.redoImage} source={iconRedo} />
          </TouchableOpacity>
        </View>
      </>
    );
  };

  const displayAnimation = ({ illustrationSource, instruction, title }) => {
    return (
      <>
        <Animation widthRatio={1} heightRatio={props.minimal ? 2 / 5 : 1 / 2} animation={illustrationSource} />
        {props.drill.steps.length > 1 && (
          <>
            <View style={styles.description}>
              <View style={styles.containerAnimation}>
                <View style={styles.descriptionAnimation}>
                  <View style={styles.subSubWrapper}>
                    <Text style={styles.fitness}>{title}</Text>
                  </View>
                  <TouchableOpacity style={styles.container} onPress={() => checkAnimationFast()}>
                    <Animated.View
                      style={[
                        {
                          opacity: opacityUnchecked,
                        },
                      ]}
                    >
                      <Image style={styles.buttonNext} source={buttonValidation} />
                    </Animated.View>
                    <Animated.View
                      style={[
                        {
                          opacity: opacityChecked,
                        },
                      ]}
                    >
                      <Image style={styles.buttonNext} source={buttonValidationGradient} />
                    </Animated.View>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
            <View style={styles.lines} />
          </>
        )}
        <Text style={styles.instruction}>{instruction}</Text>
      </>
    );
  };

  const displayYoutube = ({ illustrationSource, title, instruction }) => {
    return (
      <>
        <View style={{ height: 250 }}>
          <WebView
            source={{
              uri: illustrationSource,
            }}
          />
        </View>
        {props.drill.steps.length > 1 && (
          <>
            <View style={styles.description}>
              <View style={styles.containerAnimation}>
                <View style={styles.descriptionAnimation}>
                  <View style={styles.subSubWrapper}>
                    <Text style={styles.fitness}>{title}</Text>
                  </View>
                  <TouchableOpacity style={styles.container} onPress={() => checkAnimation()}>
                    <Animated.View
                      style={[
                        {
                          opacity: opacityUnchecked,
                        },
                      ]}
                    >
                      <Image style={styles.buttonNext} source={buttonValidation} />
                    </Animated.View>
                    <Animated.View
                      style={[
                        {
                          opacity: opacityChecked,
                        },
                      ]}
                    >
                      <Image style={styles.buttonNext} source={buttonValidationGradient} />
                    </Animated.View>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
            <View style={styles.lines} />
            <Text style={styles.instruction}>{instruction}</Text>
          </>
        )}
      </>
    );
  };

  const displayVimeo = ({ illustrationSource, repetition, title, sounds }) => {
    const isUniqueStep = props.drill.steps.length === 1;
    return (
      <>
        <View style={[{ height: 250 }, isUniqueStep && styles.videoAlone]}>
          <VimeoVideo vimeoId={illustrationSource} screenWidth={screenDimension.width} sounds={sounds} />
        </View>
        {!isUniqueStep && (
          <>
            <View style={styles.description}>
              <View style={styles.subWrapper}>
                <Text style={styles.fitness}>{repetition}</Text>
              </View>
              <View style={styles.subSubWrapper}>
                <Text style={styles.fitness}>{title}</Text>
              </View>
              <TouchableOpacity style={styles.container} onPress={() => checkAnimation()}>
                <Animated.View
                  style={[
                    {
                      opacity: opacityUnchecked,
                    },
                  ]}
                >
                  <Image style={styles.buttonNext} source={buttonValidation} />
                </Animated.View>
                <Animated.View
                  style={[
                    {
                      opacity: opacityChecked,
                    },
                  ]}
                >
                  <Image style={styles.buttonNext} source={buttonValidationGradient} />
                </Animated.View>
              </TouchableOpacity>
            </View>

            <View style={styles.lines} />
            <View style={styles.containerAnimation}>
              <View style={styles.container}>{displayNextStep()}</View>
            </View>
          </>
        )}
      </>
    );
  };

  return (
    <GestureRecognizer
      style={styles.container}
      onSwipeLeft={checkAnimationFast}
      config={swipeConfig}
      onSwipeRight={decrementStepIndex}
    >
      {checkSwitch()}
    </GestureRecognizer>
  );
};

const screenDimension = Dimensions.get('window');
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  containerFinish: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    padding: 20,
  },
  redoMessage: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.COLOR_PRIMARY,
    fontWeight: 'bold',
    marginVertical: 50,
  },
  containerAnimation: {
    flexDirection: 'row',
    alignSelf: 'flex-end',
  },
  description: {
    flexDirection: 'row',
    paddingBottom: 2,
  },
  descriptionAnimation: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  pageAnimation: {
    flex: 1,
  },
  fitness: {
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    fontSize: theme.FONT_SIZE_LARGE,
    color: theme.COLOR_PRIMARY,
    fontWeight: 'bold',
  },
  fitnessNext: {
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    fontSize: theme.FONT_SIZE_LARGE,
    color: theme.COLOR_SECONDARY,
  },
  separator: {
    height: 15,
    borderRightWidth: 1,
    borderRightColor: theme.COLOR_PRIMARY_LIGHT,
  },
  buttonNext: {
    position: 'absolute',
    right: 0,
    marginRight: 20,
    marginLeft: 20,
    marginTop: 20,
    width: 25,
    height: 25,
    borderRadius: 12.5,
    backgroundColor: theme.BACKGROUND_COLOR_BUTTON,
    borderWidth: 1,
    borderColor: theme.BORDER_COLOR_BUTTON_ACTIVE,
    alignItems: 'center',
  },
  subWrapper: {
    flexGrow: 0,
    flexShrink: 0,
    alignItems: 'center',
  },
  subSubWrapper: {
    flex: 6,
  },
  fakeWrapper: {
    width: 70,
  },
  lines: {
    borderBottomColor: '#DCDCDC',
    borderBottomWidth: 1,
  },
  wrapperFinish: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  finishMessage: {
    marginTop: 150,
    marginBottom: 20,
    marginLeft: 20,
    fontSize: theme.FONT_SIZE_LARGE,
    color: theme.COLOR_PRIMARY,
    fontWeight: 'bold',
  },
  instruction: {
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.COLOR_PRIMARY,
    justifyContent: 'center',
    alignItems: 'center',
  },
  redoButton: {
    width: 80,
    height: 80,
    borderRadius: 40,
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
  },
  redoImage: {
    width: 60,
    height: 60,
  },
  videoAlone: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
});

export default DrillIllustration;
