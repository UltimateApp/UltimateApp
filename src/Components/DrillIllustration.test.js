import React from 'react';
import renderer from 'react-test-renderer';
import animationMenageATrois from '../Fixtures/AnimationMenageATrois';

import DrillIllustration from './DrillIllustration';
import { SourceType } from './Fixtures';
import { createDrill } from '../Fixtures/TestFixtures';

describe('<DrillIllustration />', () => {
  it('renders correctly for a drill with video', () => {
    const drill = createDrill();
    const tree = renderer.create(<DrillIllustration drill={drill} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for a drill with animation', () => {
    const drill = createDrill({
      steps: [
        {
          illustrationType: SourceType.ANIMATION,
          source: animationMenageATrois,
        },
      ],
    });
    const tree = renderer.create(<DrillIllustration drill={drill} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for a drill with a video from vimeo', () => {
    const drill = createDrill({
      steps: [
        {
          id: 1,
          title: 'Rower Hamstring',
          illustrationType: 'vimeo',
          source: '407999139',
          repetition: '20',
          rest: '90s',
          instruction: 'blabla bla blabl blalb ',
        },
        {
          id: 2,
          title: 'Full Clean',
          illustrationType: 'vimeo',
          source: '406747741',
          repetition: '8',
          rest: '90s',
          instruction: 'blabla bla blabl blalb ',
        },
      ],
    });
    const tree = renderer.create(<DrillIllustration drill={drill} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for a drill with no video nor animation', () => {
    const drill = createDrill({
      steps: [],
    });
    const tree = renderer.create(<DrillIllustration drill={drill} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
