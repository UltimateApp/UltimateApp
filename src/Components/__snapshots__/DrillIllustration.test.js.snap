// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillIllustration /> renders correctly for a drill with a video from vimeo 1`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 80,
      "gestureIsClickThreshold": 10,
      "velocityThreshold": 0.3,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  onSwipeRight={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 250,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#A6A6A6"
          hidesWhenStopped={true}
          size="large"
        />
        <Text>
          Loading...
        </Text>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 250,
            "overflow": "hidden",
            "width": 750,
          }
        }
      >
        <ViewManagerAdapter_ExpoVideoView
          proxiedProperties={
            Object {
              "onError": [Function],
              "onFullscreenUpdate": [Function],
              "onLoad": [Function],
              "onLoadStart": [Function],
              "onReadyForDisplay": [Function],
              "onStatusUpdate": [Function],
              "resizeMode": "mock",
              "source": undefined,
              "status": Object {},
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingBottom": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 0,
          "flexShrink": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
            "marginTop": 20,
          }
        }
      >
        20
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginLeft": 20,
            "marginTop": 20,
          }
        }
      >
        Rower Hamstring
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#d5d5d5",
              "borderRadius": 12.5,
              "borderWidth": 1,
              "height": 25,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
              "position": "absolute",
              "right": 0,
              "width": 25,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#d5d5d5",
              "borderRadius": 12.5,
              "borderWidth": 1,
              "height": 25,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
              "position": "absolute",
              "right": 0,
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#DCDCDC",
        "borderBottomWidth": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingBottom": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexGrow": 0,
                  "flexShrink": 0,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#A6A6A6",
                    "fontSize": 20,
                    "marginBottom": 20,
                    "marginLeft": 20,
                    "marginTop": 20,
                  }
                }
              >
                8
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#A6A6A6",
                    "fontSize": 20,
                    "marginBottom": 20,
                    "marginLeft": 20,
                    "marginTop": 20,
                  }
                }
              >
                Full Clean
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": 70,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#DCDCDC",
            "borderBottomWidth": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<DrillIllustration /> renders correctly for a drill with animation 1`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 80,
      "gestureIsClickThreshold": 10,
      "velocityThreshold": 0.3,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  onSwipeRight={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "height": 667,
        },
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 55.583333333333336,
          "height": 55.583333333333336,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 337.5,
            },
            Object {
              "translateY": 40.019999999999996,
            },
          ],
          "width": 55.583333333333336,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4278759146,
              4281005464,
            ],
            "endPoint": Array [
              0,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              1,
              1,
            ],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "borderRadius": 55.583333333333336,
              "height": 55.583333333333336,
              "width": 55.583333333333336,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
            }
          }
        >
          1
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#AFAFAF",
          "borderRadius": 55.583333333333336,
          "height": 55.583333333333336,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 337.5,
            },
            Object {
              "translateY": 100.05,
            },
          ],
          "width": 55.583333333333336,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        2
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 55.583333333333336,
          "height": 55.583333333333336,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 337.5,
            },
            Object {
              "translateY": 300.15000000000003,
            },
          ],
          "width": 55.583333333333336,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4278759146,
              4281005464,
            ],
            "endPoint": Array [
              0,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              1,
              1,
            ],
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "borderRadius": 55.583333333333336,
              "height": 55.583333333333336,
              "width": 55.583333333333336,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
            }
          }
        >
          3
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 20,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 315,
            },
            Object {
              "translateY": 73.37,
            },
          ],
          "width": 20,
        }
      }
    />
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "marginLeft": 30,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "width": 30,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "width": 30,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "width": 30,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "gray",
          "bottom": 10,
          "height": 10,
          "left": 20,
          "position": "absolute",
          "width": 630,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 10,
          "height": 10,
          "left": 20,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "height": 25,
          "left": 10,
          "opacity": 1,
          "position": "absolute",
          "width": 157.5,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "height": 25,
          "left": 167.5,
          "opacity": 1,
          "position": "absolute",
          "width": 315,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "height": 25,
          "left": 482.5,
          "opacity": 1,
          "position": "absolute",
          "width": 172.5,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#232b2b",
        "fontSize": 16,
        "justifyContent": "center",
        "marginBottom": 20,
        "marginLeft": 20,
        "marginTop": 20,
      }
    }
  />
</View>
`;

exports[`<DrillIllustration /> renders correctly for a drill with no video nor animation 1`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 80,
      "gestureIsClickThreshold": 10,
      "velocityThreshold": 0.3,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  onSwipeRight={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 50,
        }
      }
    >
      You have completed the drill!
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 40,
          "height": 80,
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<DrillIllustration /> renders correctly for a drill with video 1`] = `
<View
  config={
    Object {
      "directionalOffsetThreshold": 80,
      "gestureIsClickThreshold": 10,
      "velocityThreshold": 0.3,
    }
  }
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[Function]}
  onSwipeRight={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 250,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        cacheEnabled={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        onContentProcessDidTerminate={[Function]}
        onHttpError={[Function]}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingProgress={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={[Function]}
        source={
          Object {
            "uri": "https://www.youtube.com/embed/JkVHrA5o23o",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            undefined,
          ]
        }
        useSharedProcessPool={true}
      />
    </View>
  </View>
</View>
`;
